File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/RenderedIdentityObject.java
package com.ulyp.ui.renderers;

import com.ulyp.core.printers.IdentityObjectRepresentation;
import com.ulyp.ui.RenderSettings;
import com.ulyp.ui.util.ClassNameUtils;
import com.ulyp.ui.util.StyledText;

import java.util.Arrays;

import static com.ulyp.ui.util.CssClass.CALL_TREE_IDENTITY_REPR;
import static com.ulyp.ui.util.CssClass.CALL_TREE_TYPE_NAME;

public class RenderedIdentityObject extends RenderedObject {

    public RenderedIdentityObject(IdentityObjectRepresentation repr, RenderSettings renderSettings) {
        super(repr.getType());

        String className = renderSettings.showTypes() ? repr.getType().getName() : ClassNameUtils.toSimpleName(repr.getType().getName());

        super.getChildren().addAll(
                Arrays.asList(
                        StyledText.of(className, CALL_TREE_TYPE_NAME),
                        StyledText.of("@", CALL_TREE_IDENTITY_REPR),
                        StyledText.of(Integer.toHexString(repr.getHashCode()), CALL_TREE_IDENTITY_REPR)
                )
        );
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/RenderedIdentityObject.kt
package com.ulyp.ui.renderers

import com.ulyp.core.recorders.IdentityObjectRecord
import com.ulyp.ui.RenderSettings
import com.ulyp.ui.util.ClassNameUtils.toSimpleName
import com.ulyp.ui.util.CssClass
import com.ulyp.ui.util.StyledText.of

class RenderedIdentityObject(record: IdentityObjectRecord, renderSettings: RenderSettings) : RenderedObject(record.type) {

    init {
        val className = if (renderSettings.showTypes()) record.type.name else toSimpleName(record.type.name)

        super.getChildren().addAll(
            listOf(
                of(className, CssClass.CALL_TREE_TYPE_NAME),
                of("@", CssClass.CALL_TREE_IDENTITY, CssClass.CALL_TREE_IDENTITY_HASH_CODE),
                of(Integer.toHexString(record.hashCode), CssClass.CALL_TREE_IDENTITY)
            )
        )
    }
}